### DISCUSSION TITLE
Slow shutdown

### DISCUSSION THREAD

**Discussion by mikiTesf** [mockoon/mockoon#630] [General]
Why does it take more than 5 seconds for the app to shutdown?

**Comment by 255kb** [comment]
It depends, the app is mostly waiting for environment to be saved which can take several seconds. I didn't experience any closing time above 1 or 2 seconds. 
If you have specific steps to reproduce this behavior let me know.

**Reply by mikiTesf** [reply]
I'm assuming you mean saving newly added routes or similar operations when you say:
> waiting for environment to be saved

To make sure if that could be what's causing the delays, I opened Mockoon just a minute ago and clicked on the window's close button without making any changes. It took 8 seconds. I'm 100% sure I don't have hardware problems. I'm running an ASUS Rog Strix with 16 Gigs of RAM and a 12 core 8th gen processor. Is there a debug output I can share that might help?

**Reply by 255kb** [reply]
There is no relevant debug to share. 
But I think the problem could come from the fact that the app is waiting for any data saving to finish but also for the telemetry session to be sent to the server. I guess if the network is slow it could take a while. 
I will change things a bit so it does not wait for the telemetry anymore. 
Thank you for reporting!

**Reply by mikiTesf** [reply]
Hmm...I haven't enabled telemetry reporting.

**Reply by 255kb** [reply]
Oh really? So I definitely don't know where this is coming from. 
I am able to locally simulate slow network and slow file saving to have a slow shutdown, but appart from these two types of events is see no other reason for the app to behave like this.

**Reply by mikiTesf** [reply]
Strange. We'll see if anyone else experiences a similar problem and look further into it. On another note, I was thinking if the option to copy the fully constructed URL (`scheme + host + API-Prefix + route`) of a route can be added to the right-click context options. I often find myself typing out the full path to routes here and there.

**Reply by 255kb** [reply]
That could be an interesting option. It could also be a copy button next to the path input. Feel free to open an issue to track this feature request.
